app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,42 @@
+class PostsController < ApplicationController
+    before_action :authenticate_user!
+
+    def create
+        @post = current_user.posts.build(post_params)
+        if @post.save
+            redirect_back(fallback_location: root_path)
+            flash[:notice] = "Post successfully created"
+        else
+            flash[:error] = @post.errors.full_messages
+            redirect_back(fallback_location: root_path)
+        end
+    end
+
+    def destroy
+        @post = Post.find(params[:id])
+		@post.destroy
+        redirect_back(fallback_location: root_path)
+        flash[:notice] = "Post successfully deleted"
+    end
+
+    def edit
+        @post = Post.find(params[:id])
```

Comment:
we are doing this same `@post = Post.find(params[:id])` for edit, update, destroy and will do it for show. let's consider making a private method and setting it with a before action.
[OPTIONAL]